File: firebase-ai-friendly-meals/android/app/src/main/java/com/google/firebase/example/friendlymeals/data/repository/AIRepository.kt
package com.google.firebase.example.friendlymeals.data.repository

import android.graphics.Bitmap
import com.google.firebase.example.friendlymeals.data.datasource.AIRemoteDataSource
import com.google.firebase.example.friendlymeals.data.schema.MealSchema
import com.google.firebase.example.friendlymeals.data.schema.RecipeSchema
import javax.inject.Inject

class AIRepository @Inject constructor(
    private val aiRemoteDataSource: AIRemoteDataSource
) {
    suspend fun generateIngredients(image: Bitmap): String {
        return aiRemoteDataSource.generateIngredients(image)
    }

    suspend fun generateRecipe(ingredients: String, notes: String): RecipeSchema {
        return aiRemoteDataSource.generateRecipe(ingredients, notes)
    }

    suspend fun generateRecipePhoto(recipeTitle: String): Bitmap? {
        return aiRemoteDataSource.generateRecipePhoto(recipeTitle)
    }

    suspend fun generateRecipePhotoImagen(recipeTitle: String): Bitmap? {
        return aiRemoteDataSource.generateRecipePhotoImagen(recipeTitle)
    }

    suspend fun scanMeal(image: Bitmap): MealSchema {
        return aiRemoteDataSource.scanMeal(image)
    }
}

File: firebase-ai-friendly-meals/flutter/flutter-workshop-starter/lib/data/repository/ai_repository.dart
import 'dart:typed_data';
import 'package:firebase_ai_friendly_meals/data/datasource/ai_remote_data_source.dart';
import 'package:injectable/injectable.dart';

@injectable
class AIRepository {
  final AIRemoteDataSource _aiRemoteDataSource;

  AIRepository(this._aiRemoteDataSource);

  Future<String> generateIngredients(Uint8List image) async {
    return _aiRemoteDataSource.generateIngredients(image);
  }

  Future<String> generateRecipe(String ingredients, String notes) async {
    return _aiRemoteDataSource.generateRecipe(ingredients, notes);
  }

  Future<Uint8List> generateRecipeImage(String recipe) async {
    return _aiRemoteDataSource.generateRecipeImage(recipe);
  }
}
